# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/basetypes/HyperTextAccessibleBase.h
# Commit: e12dbde5824b
# Full Hash: e12dbde5824b03fe31e2ce71d183edcc162cc2e9
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-05-03 09:42:08
# Regressor Bug: 1766794
# File Overlap Count: 2
# Description:
#   Bug 1766794: Lazily cache HyperText offsets for RemoteAccessibles.
#   
#   We already have a similar cache in local HyperTextAccessible.
#   Unfortunately, we can't use common code here because we don't want to waste memory having a member variable on all RemoteAccessibles, but local HyperTextAccessibles don't have mCachedFields.
#   This improves performance significantly when walking the text attributes of a container with a large number of text leaf children, such as is encountered when using view source.
# ==============================================================================

diff -r 6a2ad49b2630 -r e12dbde5824b accessible/basetypes/HyperTextAccessibleBase.h
--- a/accessible/basetypes/HyperTextAccessibleBase.h	Mon May 02 22:38:07 2022 +0000
+++ b/accessible/basetypes/HyperTextAccessibleBase.h	Mon May 02 22:40:43 2022 +0000
@@ -45,7 +45,7 @@
    *
    * @param  aOffset  [in] the given text offset
    */
-  virtual int32_t GetChildIndexAtOffset(uint32_t aOffset) const;
+  virtual int32_t GetChildIndexAtOffset(uint32_t aOffset) const = 0;
 
   /**
    * Return child accessible at the given text offset.
@@ -69,7 +69,7 @@
    * Return text offset for the child accessible index.
    */
   virtual int32_t GetChildOffset(uint32_t aChildIndex,
-                                 bool aInvalidateAfter = false) const;
+                                 bool aInvalidateAfter = false) const = 0;
 
   /**
    * Return character count within the hypertext accessible.